x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -127,60 +123,44 @@ mod tests {
 
```

Comment:
Does it still make sense for the plan responses possibly error? Shouldn't the whole plan either pass or fail since we can be as granular as we want? 

It seems like making sure all the plan responses aren't errors should just be part of the `run_plan` method. 

If you create a new issue (if you use `create issue from comment`, please add extra context), link the new issue here, then resolve this comment. No need to make changes in this PR.

Reply:
Actually, I think the whole `run_plan` method should probably removed since we only run one step at a time synchronously anyway.

Reply:
I will not create the issue then, since we will remove this idea of plan in a followup PR